toqito/matrices/comparison.py

```diff
@@ -0,0 +1,49 @@
+"""Computes comparison matrix."""
+
+import numpy as np
+
+
+def comparison(mat: np.ndarray) -> np.ndarray:
+    r"""Compute comparison matrix of a given square matrix.
+
+    This function computes the comparison matrix :math:`M(A)` for a square matrix :math:`A` as defined in
+    :cite:`WikiComparisonMatrix`. For each entry, the diagonal entries are given by the absolute value of the original
+    diagonal entries of :math:`A`, while the off-diagonal entries are given by minus the absolute value of the
+    corresponding entries. In other words,
+
+    .. math::
+        m_{ij} =
+        \begin{cases}
+        |a_{ij}|, & \text{if } i = j, \\
+        -|a_{ij}|, & \text{if } i \neq j.
+        \end{cases}
+
+    Examples
+    ========
+    >>> import numpy as np
+    >>> from matrices.comparison import comparison_matrix
+    >>> A = np.array([[2, -1],
+    ...               [3, 4]])
+    >>> comparison_matrix(A)
+    array([[ 2., -1.],
+        [-3.,  4.]])
```

Comment:
You can use jupyter execute code block here.

Reply:
Done in 03f7ce3f25218078284cf129afc72c151ad9bde8